src/content/guides/migrating.md

```diff
@@ -16,12 +16,117 @@ contributors:
   - ndelangen
 ---
 
-The following sections describe the major changes from webpack 1 to 2.
+This guide only shows major changes that affect end users. For more details please see [the changelog](https://github.com/webpack/webpack/releases).
+
+# Migration from 3 to 4
+
+[Changelog](https://github.com/webpack/webpack/releases/tag/v4.0.0)
+
+## node.js 4
+
+If you are still using node.js 4 or lower, you need to upgrade your node.js installation
+
+## CLI
+
+The CLI has moved to a separate package: webpack-cli. You need to install it to use the CLI.
+
+``` text
+yarn add webpack-cli -D
+```
+
+``` text
+npm install webpack-cli -D
+```
+
+## Update plugins
+
+Many 3rd-party plugins need to be upgraded to their latest version to be compatible.
+
+## mode
 
-T> Note that there were far fewer changes between 2 and 3, so that migration shouldn't be too bad. If you are running into issues, please see [the changelog](https://github.com/webpack/webpack/releases) for details.
+Add the new `mode` option to your config. Set it to production or development in your configuration depending on config type.
 
-W> This content may be moved to the blog post in the near future as version 2 has been out for a while. On top of that, version 3 was recently released and version 4 is on the horizon. As noted above, folks should instead to refer to [the changelog](https://github.com/webpack/webpack/releases) for migrations.
+``` diff
++mode: "production",
+```
+
+``` diff
++mode: "development",
+```
+
+Alternativly you can pass it via CLI: `--mode production`/`--mode development`
+
+## Deprecated/Removed plugins
 
+These plugins can be removed from configuration as they are default in production mode:
+
+``` diff
+-new NoEmitOnErrorsPlugin(),
+-new ModuleConcatenationPlugin(),
+-new DefinePlugin({ "process.env.NODE_ENV": JSON.stringify("production") })
+-new UglifyJsPlugin()
+```
+
+These plugins are default in development mode
+
+``` diff
+-new NamedModulesPlugin()
+```
+
+These plugins were deprecated and are now removed:
+
+``` diff
+-new NoErrorsPlugin(),
+-new NewWatchingPlugin(),
+```
+
+## CommonsChunkPlugin
+
+The `CommonsChunkPlugin` was removed. Instead the `optimization.splitChunks` options can be used.
+
+See documentation for `optimization.splitChunks` to details. The default configuration may already suit your needs.
+
+When generating the HTML from the stats you can use `optimization.splitChunks.chunks: "all"` which is the optimal configuration in most cases.
+
+## import() and CommonJS
+
+When using `import()` to load non-ESM the result has changed in webpack 4. You now need access the `default` property to get the value of `module.exports`.
```

Comment:
Now you need to access the `default` property to get the value of `module.exports`.
__non-esm.js__
```javascript
module.exports = {
  sayHello: () => { console.log('hello world') }
};
```

__example.js__
```javascript
function sayHello() {
  import('./non-esm.js').default.sayHello();
}
sayHello();
```